src/native/eventpipe/ep.c

```diff
@@ -611,7 +611,7 @@ write_event (
 	EP_ASSERT (payload != NULL);
 
 	// We can't proceed if tracing is not initialized.
-	ep_return_void_if_nok (ep_volatile_load_eventpipe_state () == EP_STATE_INITIALIZED);
+	ep_return_void_if_nok (ep_volatile_load_eventpipe_state () >= EP_STATE_INITIALIZED);
```

Comment:
```suggestion
	ep_return_void_if_ok (ep_volatile_load_eventpipe_state () == EP_STATE_NOT_INITIALIZED);
```

might be closer to the [original incarnation](https://github.com/dotnet/runtime/blob/49f4fb85df7339eddd07ce4b6065d370cd93d085/src/coreclr/src/vm/eventpipe.cpp#L755-L757), but what you have is also correct.

Reply:
Yes, I aligned to the pattern used for this check elsewhere, but either one works.